src/components/common/DataTable.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <div class="table-responsive">
+    <table :class="darkMode ? 'dark-table' : 'table'">
+      <thead>
+        <tr :class="darkMode ? 'table-header-dark' : 'table-header-light'">
+          <th
+            @click="setSortBy(column)"
+            v-for="(column, index) in fields"
+            :key="index"
+            scope="col"
+            :class="isColumnSort(column) ? 'cursor' : ''"
+            :style="getWidthStyle(column)"
+          >
+            <slot :name="`head(${column.key})`">
+              {{ column.label }}
+            </slot>
+            <font-awesome-icon
+              v-if="column.tooltip"
+              icon="info-circle"
+              class="mr-1"
+              v-b-popover.hover.top="column.tooltip"
+            />
+            <font-awesome-icon
+              v-if="column.key === sortBy"
+              :icon="descOrder ? 'caret-down' : 'caret-up'"
+              :class="darkMode ? 'text-white' : 'text-primary'"
+            />
+          </th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="pool in paginatedItems" :key="pool.id">
+          <td v-for="(column, index) in fields" :key="index">
+            <slot
+              :name="`cell(${column.key})`"
+              :item="pool"
+              :value="pool[column.key]"
+            >
+              {{ pool[column.key] }}
+            </slot>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+
+    <table-pagination
+      v-if="!hidePagination"
+      :current-page.sync="currentPage"
+      :rowCount="modifiedItems.length"
```

Comment:
Now that we're adding linting stuff we're locking down on props, in this case I think it's meant to be kebab-case. 

https://vuejs.org/v2/style-guide/#Prop-name-casing-strongly-recommended

Reply:
fixed